### ISSUE TITLE
TBSync EAS not working since 2-3 days

### ISSUE THREAD

**Issue by oe3gwu** [jobisoft/TbSync#706]
## Your environment
Kubuntu 23.10
Thunderbird from Repos with installed TBSync

TbSync + EAS Provider version: 4.8
Thunderbird version: 115.6.0

## Expected behavior
TBSync should sync my Hotmail Account

## Actual behavior
Not working. Always get the error to enter my password.

## Steps to reproduce
Use an Hotmail Account with TBSync

Logs:
```
** Thu Jan 25 2024 18:38:19 GMT+0100 (Mitteleuropäische Normalzeit) **
[setSyncState] : State: prepare.request.options, Account: rainer_weninger@hotmail.com
 ** Thu Jan 25 2024 18:38:19 GMT+0100 (Mitteleuropäische Normalzeit) **
[Sending] : OPTIONS https://eas.outlook.com/Microsoft-Server-ActiveSync
 ** Thu Jan 25 2024 18:38:19 GMT+0100 (Mitteleuropäische Normalzeit) **
[setSyncState] : State: send.request.options, Account: rainer_weninger@hotmail.com
 ** Thu Jan 25 2024 18:38:19 GMT+0100 (Mitteleuropäische Normalzeit) **
[setSyncState] : State: eval.request.options, Account: rainer_weninger@hotmail.com
 ** Thu Jan 25 2024 18:38:19 GMT+0100 (Mitteleuropäische Normalzeit) **
[EAS OPTIONS with response (status: 200)] : 
responseText: 
responseHeader(MS-ASProtocolVersions): 2.5,14.0,14.1,16.0,16.1
responseHeader(MS-ASProtocolCommands):
Sync,SendMail,SmartForward,SmartReply,GetAttachment,FolderSync,FolderCreate,FolderD
lete,FolderUpdate,MoveItems,GetItemEstimate,MeetingResponse,Search,Settings,Ping,Ite
Operations,ResolveRecipients,Find
 ** Thu Jan 25 2024 18:38:19 GMT+0100 (Mitteleuropäische Normalzeit) **
[setSyncState] : State: prepare.request.setdeviceinfo, Account:
rainer_weninger@hotmail.com
 ** Thu Jan 25 2024 18:38:19 GMT+0100 (Mitteleuropäische Normalzeit) **
[setSyncState] : State: send.request.setdeviceinfo, Account: rainer_weninger@hotmail.com
 ** Thu Jan 25 2024 18:38:19 GMT+0100 (Mitteleuropäische Normalzeit) **
[Sending (EAS v14.0)] : POST https://eas.outlook.com/Microsoft-Server-ActiveSync
Cmd=Settings&User=rainer_weninger%40hotmail.com&DeviceType=TbSync&DeviceId=MZ
B559bf278587b0b4abbdbb92cb4ec
 ** Thu Jan 25 2024 18:38:19 GMT+0100 (Mitteleuropäische Normalzeit) **
[setSyncState] : State: passwordprompt, Account: rainer_weninger@hotmail.com
 ** Thu Jan 25 2024 18:38:21 GMT+0100 (Mitteleuropäische Normalzeit) **
[setSyncState] : State: send.request.setdeviceinfo, Account: rainer_weninger@hotmail.com
 ** Thu Jan 25 2024 18:38:21 GMT+0100 (Mitteleuropäische Normalzeit) **
[Sending (EAS v14.0)] : POST https://eas.outlook.com/Microsoft-Server-ActiveSync
Cmd=Settings&User=rainer_weninger%40hotmail.com&DeviceType=TbSync&DeviceId=MZB559bf278587b0b4abbdbb92cb4ec
 ** Thu Jan 25 2024 18:38:21 GMT+0100 (Mitteleuropäische Normalzeit) **
[setSyncState] : State: passwordprompt, Account: rainer_weninger@hotmail.com
 ** Thu Jan 25 2024 18:38:23 GMT+0100 (Mitteleuropäische Normalzeit) **
[EventLog] : 401
  ** Thu Jan 25 2024 18:38:23 GMT+0100 (Mitteleuropäische Normalzeit) **
[EventLog] : 401
  ** Thu Jan 25 2024 18:38:23 GMT+0100 (Mitteleuropäische Normalzeit) **
[setSyncState] : State: accountdone, Account: rainer_weninger@hotmail.com
```

**Comment by Jumpiebeen** [comment]
Is there any way to roll this back? I have the same issue for the same time period.

**Comment by jobisoft** [comment]
Microsoft apparently dropped support for plain password auth. You need to use the OAUTH option (3rd option when creating an EAS account).

**Comment by jcs0596223** [comment]
I am having the same problem - my password is not being accepted but I can read/send email via IMAP

**Comment by Jumpiebeen** [comment]
As John said, you should delete the TBSync account and re-create it using the OATH option for authentication. On my version the OATH is labled Office365.
![Screenshot from 2024-01-27 08-25-32](https://github.com/jobisoft/TbSync/assets/31117449/a4fdc5eb-465c-4309-b4ba-5b59438d50c1)

**Comment by jcs0596223** [comment]
Thanks - that appears to have done the trick. It was not at all obvious!

**Comment by Dave-Lowndes** [comment]
Deleting/re-creating as the Microsoft Office 365 setting worked for 1 of my 3 outlook.com accounts.
The other 2 fail with:

> The server does not provide information about the supported ActiveSync versions....

Setting the account to a specific AS version results in "Communication error (HTTP status 503)."
Caching issue maybe?

**Comment by oe3gwu** [comment]
OAuth worked for me. Thank you!

**Comment by oe3gwu** [comment]
Short update. I still get an error.

![grafik](https://github.com/jobisoft/TbSync/assets/17622063/92c97690-9672-43f5-8032-a707b86ffb8d)

**Comment by oe3gwu** [comment]
It works for me to use TBSync with OAuth2. To all that read this and have IMAP issues too, you need to use OAuth2 also on the IMAP Settings.

**Comment by hugo-b-r** [comment]
https://support.mozilla.org/en-US/kb/microsoft-oauth-authentication-and-thunderbird-202?redirectslug=microsoft-oauth-authentication-and-thunderbird-202-redirect-1&redirectlocale=en-US

**Comment by Dave-Lowndes** [comment]
> Deleting/re-creating as the Microsoft Office 365 setting worked for 1 of my 3 outlook.com accounts. The other 2 fail with:
> 
> > The server does not provide information about the supported ActiveSync versions....

Today, 2 of 3 work with TbSync.
Maybe the third will tomorrow!

**Comment by CharalamposKarampasis** [comment]
> Microsoft apparently dropped support for plain password auth. You need to use the OAUTH option (3rd option when creating an EAS account).

Hi all,
A quick question, which of the following steps should I do when I am changing the OAUTH ? See attached photos
![Screenshot from 2024-01-29 16-57-41](https://github.com/jobisoft/TbSync/assets/158076785/9c59952e-3437-47a7-a64b-b8975e848906)
![Screenshot from 2024-01-29 16-58-46](https://github.com/jobisoft/TbSync/assets/158076785/7f96525b-6e02-4b22-8161-6d155ab2c02b)

**Comment by CharalamposKarampasis** [comment]
> SSL/TLS is what I'm using (successfully)

You had right , all done and sync!! At the beginning a had a message ''error''again but no matter I just click to continue and fixed.

**Comment by Dave-Lowndes** [comment]
I have 3 Outlook accounts configured and I'm finding that only the first synchronizes OK, others have failures (HTTP 503).

However, if I close Thunderbird, re-open it and then quickly open TbSync and use "Synchronize now" on say the 3'rd account it works, and now the first and second have the failures. I can now repeatedly use "Synchronize now" OK on the 3'rd account, and the other 2 repeatedly fail the same way.

Is something from the initial sync connection being held open so the others then fail?

**Comment by jobisoft** [comment]
That is related to a missing cookie implementation/support in TbSync. I somehow need to give all connections an independent cookie stores, otherwise they all try to use the same and that does not work. I am looking into this.

**Comment by goshoom** [comment]
I couldn't get it working since January, but I finally succeeded. I was connecting to outlook.com and the primary alias of my Microsoft account wasn't the one with outlook.com address. When I made that primary, TB sync has been able to connect.
It's looks weird to me that I suddenly had to change the primary alias (to a different one than I actually want and used for years), but at least there is a solution.
